testcase.py

```diff
@@ -361,11 +372,22 @@ def set_forecast_parameters(self,horizon,interval):
         None
         
         '''
+
+        try: 
+        
+             self.horizon = float(horizon)
         
-        self.horizon = float(horizon)
-        self.interval = float(interval)
+             self.interval = float(interval)
+            
+        except TypeError:
+        
+             return {'message':'failure','error':'insufficient input','result':None}
+             
+        except ValueError:
         
-        return None
+             return {'message':'failure','error':'invalid input','result':None}
```

Comment:
It is not clear what the problem is with the input. A better description is needed. I don't think ``'invalid'`` is the correct word here.  It aligns more with ``TypeError``.